### PR TITLE
Homework10: Yanislav Terentev

### PR DESCRIPTION
**Pull request by YanislavTerentev**

### CODE REVIEW

File: Homework/Homework_10/cache_func.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+def cache(func):
+    cache_dictionary = {}
+
+    def wrapper(arg):
+        if arg in cache_dictionary:
+            return cache_dictionary[arg]
+        cache_dictionary[arg] = func(arg)
+        return func(arg)
```
Comment (Stml89) [inline Homework/Homework_10/cache_func.py]:
```suggestion
        return cache_dictionary[arg]
```

Comment (Stml89) [reply Homework/Homework_10/cache_func.py]: you do not need to call the same function twice

File: Homework/Homework_10/return_number.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        return result
+    return wrapper
+
+
+@is_number
+def checking_strung(sting):
+    try:
+        float(sting)
+        s = float(sting)
+        return s
```
Comment (Stml89) [inline Homework/Homework_10/return_number.py]:
```suggestion
        return  float(sting)
```

File: Homework/Homework_10/return_number.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+ERROR_MESSAGE = 'Error! Function result is not number.'
+
+
+def is_number(func):
+    def wrapper(sting):
+        result = func(sting)
+        if not isinstance(result, float):
+            print(ERROR_MESSAGE)
```
Comment (Stml89) [inline Homework/Homework_10/return_number.py]: remove this `print()`

File: Homework/Homework_10/type_decorator.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+def typed(type_element):
+    def convert(func):
+        def wrapper(*args):
+            formatted_array = []
+            for arg in args:
+                if isinstance(arg, type_element):
+                    formatted_array.append(arg)
+                else:
+                    value = type_element(arg)
+                    formatted_array.append(value)
```
Comment (Stml89) [inline Homework/Homework_10/type_decorator.py]:
```suggestion
                    formatted_array.append(type_element(arg))
```

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: Homework/Homework_10/cache_func.py (hunk 1 above)
Comment (YanislavTerentev) [reply Homework/Homework_10/cache_func.py]: Done

File: Homework/Homework_10/return_number.py (hunk 2 above)
Comment (YanislavTerentev) [reply Homework/Homework_10/return_number.py]: Done

File: Homework/Homework_10/return_number.py (hunk 3 above)
Comment (YanislavTerentev) [reply Homework/Homework_10/return_number.py]: Done

File: Homework/Homework_10/type_decorator.py (hunk 4 above)
Comment (YanislavTerentev) [reply Homework/Homework_10/type_decorator.py]: Done

**Review by Stml89 — APPROVED** [review]

**Pull request merged**